### PR TITLE
acr: warn on using registry name with uppercase

### PR DESCRIPTION
**Pull request by yugangw-msft**
- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by fengzhou-msft** [comment]
@jiasli you get the chance to review yugang's code.

**Comment by yonzhan** [comment]
@jiasli a good opportunity for you to deal with this PR:)

File: src/azure-cli/azure/cli/command_modules/acr/custom.py [hunk 1]
```diff
@@ -141,6 +141,12 @@ def acr_login(cmd,
         username=username,
         password=password)
 
+    # warn casing difference caused by ACR normalizes to lower on login_server
+    parts = login_server.split('.')
+    if registry_name != parts[0] and registry_name.lower() == parts[0]:
+        logger.warning('Uppercase characters are detected in the registry name. When use its server url in docker'
+                       ' commands, to avoid authentication errors, use all lowercase.')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/acr/custom.py]: Grammar error: When **using**...

File: src/azure-cli/azure/cli/command_modules/acr/custom.py [hunk 2]
```diff
@@ -141,6 +141,12 @@ def acr_login(cmd,
         username=username,
         password=password)
 
+    # warn casing difference caused by ACR normalizes to lower on login_server
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/acr/custom.py]: normalizes -> normalizing

**Pull request merged**